# interpretation-of-linearregression

## Summary

* Objective
* Modeling
* Interpretation

## Objective

I would like to know what an interpretation of `LinearRegression` is like. This model was made with price, season and weight of orange juice. I set the price as a target value, and the others as predictors so I could know how the season and weight determine the price of orange juice. Here, assume that season is when the orange juices were sold.

## Modeling

In [7]:
# Create dataset
import pandas as pd
import numpy as np

np.random.seed(42)
weight = [550, 500, 400, 900, 350, 800, 450, 300, 600]
season = ['Summer', 'Spring', 'Fall', 'Winter']
orange_juice = pd.DataFrame()
orange_juice['weight'] = [np.random.choice(weight) for _ in range(50)]
orange_juice['season'] = [np.random.choice(season) for _ in range(50)]
orange_juice['price'] = round((1/100)*orange_juice['weight']*orange_juice['season'].map({'Spring': 1.2, 'Summer': 2, 'Fall':1.4 , 'Winter': 1}) + 10, 1)
orange_juice.head()

weight  season  price
0     450  Winter   14.5
1     900  Spring   20.8
2     300  Spring   13.6
3     350  Spring   14.2
4     450  Spring   15.4

In [8]:
# Save the dataset as csv
# orange_juice.to_csv('../data/orange_juice.csv', index=False)

In [99]:
# Modeling with LinearRegression
lr = LinearRegression()

features = orange_juice.drop(columns='price')
features = pd.get_dummies(data=features, columns={'season'}, drop_first=True)

In [100]:
# Get coefficient of each column
lr.fit(features, orange_juice['price'])

[(column, coefficient) for column, coefficient in zip(features.columns, lr.coef_)]

[('weight', 0.01412144030119758),
 ('season_Spring', -1.0143739736452113),
 ('season_Summer', 3.0573637201105446),
 ('season_Winter', -1.8553811030560343)]

## Interpretation

> For each increase of weight unit, we can expect the price is increased by \\$0.014 holding all other features consistent
Also
We can expect price of the orange juice in Summer is \\$3.05 higher than its price in Fall holding all other features consistent